bpf/kprobe_pwru.c

```diff
@@ -338,4 +355,19 @@ int kprobe_skb_lifetime_termination(struct pt_regs *ctx) {
 	return 0;
 }
 
+SEC("fentry/tc")
+int BPF_PROG(fentry_tc, struct sk_buff *skb) {
+	struct event_t event = {};
+
+	if (!handle_everything(skb, ctx, &event))
+		return BPF_OK;
+
+	event.type = EVENT_TYPE_TC;
+	event.skb_addr = (u64) skb;
+	event.addr = BPF_PROG_ID;
```

Comment:
Do you think it's possible to use `bpf_get_func_ip` here to fetch rip? So we don't have to intro a new event field `type` and can re-use the existing `addr2name` to look up for prog name.

Reply:
At the beginning, I retrieved its `addr` by `bpf_get_func_ip()`. But it didn't show the corresponding name expectedly.

I'll confirm it again.

BTW, `type` is an unused legacy field of `event`.

Reply:
I confirm that `bpf_get_func_ip()` is helpless for tracing bpf progs.

When change to

```
	// event.type = EVENT_TYPE_TC;
	event.skb_addr = (u64) skb;
	event.addr = bpf_get_func_ip(ctx);
```

then the output is

```
#  ./pwru --output-limit-lines 10  --filter-trace-tc --filter-func '.*udp' icmp
2023/10/20 13:18:06 Attaching kprobes (via kprobe-multi)...
1 / 1 [------------------------------------------------------------------------------------------------------------] 100.00% ? p/s
2023/10/20 13:18:06 Attached (ignored 0)
2023/10/20 13:18:06 Listening for events..
               SKB    CPU          PROCESS                     FUNC
0xffff940942620200      3           [pwru]       0xffffffffc031b788
0xffff940942620500      3        [<empty>]       0xffffffffc031b788
0xffff940942620c00      3        [<empty>]       0xffffffffc031b788
0xffff94093ef61100      3        [<empty>]       0xffffffffc031b788
0xffff94093ef61500      3           [pwru]       0xffffffffc031b788
0xffff94088327ea00      3        [<empty>]       0xffffffffc031b788
0xffff940942620d00      3        [<empty>]       0xffffffffc031b788
0xffff940942620000      3        [<empty>]       0xffffffffc031b788
0xffff940942620700      3        [<empty>]       0xffffffffc031b788
0xffff940942620a00      3           [pwru]       0xffffffffc031b788
2023/10/20 13:18:10 Printed 10 events, exiting program..
```

Reply:
I checked locally and realize it's not the problem of `bpf_get_func_ip()`, it's pwru that fails to collect all symbols and addresses from /proc/kallsyms:

```
	addr2name, err := pwru.GetAddrs(funcs, flags.OutputStack || len(flags.KMods) != 0)
```

By default the second argument has `false` value so symbols like `ffffffffc0147cac t bpf_prog_5269f21b11aec316_on_egress	[bpf]` isn't taken into account.

I'm okay to have current PR merged, then I can clean this up on my own.

Reply:
Thank Gray, I change the 2nd argument of `pwru.GetAddrs()` as your suggestion.

So, now, it can output:

```
# ./pwru --output-limit-lines 10  --filter-trace-tc --filter-func '.*udp' icmp
2023/10/21 07:35:01 Attaching kprobes (via kprobe-multi)...
1 / 1 [------------------------------------------------------------------------------------------------------------] 100.00% ? p/s
2023/10/21 07:35:01 Attached (ignored 0)
2023/10/21 07:35:01 Listening for events..
               SKB    CPU          PROCESS                     FUNC
0xffff94094484ae00      3        [<empty>]                    dummy
0xffff94094484a900      3        [<empty>]                    dummy
0xffff94094484a400      3        [<empty>]                    dummy
0xffff94088333ff00      3        [<empty>]                    dummy
0xffff94088333f700      3        [<empty>]                    dummy
0xffff94088333f900      3        [<empty>]                    dummy
0xffff94088333fb00      3        [<empty>]                    dummy
0xffff94088333f600      3        [<empty>]                    dummy
0xffff94088333f500      3        [<empty>]                    dummy
0xffff94088333fe00      3        [<empty>]                    dummy
2023/10/21 07:35:05 Printed 10 events, exiting program..
```